integration/kube_integration_test.go

```diff
@@ -1573,21 +1614,39 @@ func testKubeJoin(t *testing.T, suite *KubeSuite) {
 	// created. Sadly though the k8s API doesn't give us much indication of when that is.
 	time.Sleep(time.Second * 5)
 
-	participantStdinR, participantStdinW, err := os.Pipe()
-	participantStdoutR, participantStdoutW, err := os.Pipe()
+	participantStdinR, participantStdinW := io.Pipe()
+	participantStdoutR, participantStdoutW := io.Pipe()
 
-	tc, err := teleport.NewClient(helpers.ClientConfig{})
+	tc, err := teleport.NewClient(helpers.ClientConfig{
+		Login:   hostUsername,
+		Cluster: helpers.Site,
+		Host:    Host,
+	})
 	require.NoError(t, err)
 
 	tc.Stdin = participantStdinR
 	tc.Stdout = participantStdoutW
 
+	var session types.SessionTracker
+	require.Eventually(t, func() bool {
```

Comment:
nit: i think this `require.Eventually(` can replace the `time.Sleep(time.Second * 5)` above on line 1615 to save a few seconds.